### PR TITLE
Rename Equinox.Stream -> Decider

### PR DESCRIPTION
**Pull request by bartelink**
Finally, a nice name for it; based on thinking shared by @thinkbeforecoding on DDD-CQRS-ES slack

### CODE REVIEW

**Comment by thinkbeforecoding** [comment]
To give more information about the reason of the name:
- Stream is linked the the specificity of the storage in an Event Sourcing system
- Aggregate (in DDD terminology) is too much about the shape (which tend to disappear in an Event Sourcing system)

Finally the the role of this is to take decision (in autonomie, not relying on other information).
it signature is:
decide : Command -> State -> Event[]
evolve: State -> Event -> State
initialState: State

it can be made a pure decider by having State = Event[]:
initialState: []
evolve: State -> Event -> Event[]  // just append event to list

and the decider becomes (but avoid it, it's less efficient at runtime, but it's equivalent conceptually):
decide: Command -> history:Event[] -> Event[]

Here is the reason to call it Decider

**Comment by bartelink** [comment]
I'm intending to merge this prior to the first RC release of V3

File: DOCUMENTATION.md [hunk 1]
```diff
@@ -696,9 +696,9 @@ elements you'll touch in a normal application are:
   internal implementation of Optimistic Concurrency Control / retry loop used
   by `Stream`. It's recommended to at least scan this file as it defines the
```
Comment (thinkbeforecoding) [inline DOCUMENTATION.md:697]: You left one here 😉

File: DOCUMENTATION.md [hunk 2]
```diff
@@ -696,9 +696,9 @@ elements you'll touch in a normal application are:
...
   Transaction semantics everything is coming together in service of.
-- [`type Stream`](https://github.com/jet/equinox/blob/master/src/Equinox/Equinox.fs#L11) -
+- [`type Decider`](https://github.com/jet/equinox/blob/master/src/Equinox/Decider.fs#L11) -
   surface API one uses to `Transact` or `Query` against a specific stream
-- [`type Target` Discriminated Union](https://github.com/jet/equinox/blob/master/src/Equinox/Equinox.fs#L42) -
+- [`type Target` Discriminated Union](https://github.com/jet/equinox/blob/master/src/Equinox/Decider.fs#L42) -
   used to identify the Stream pertaining to the relevant Aggregate that
   `resolve` will use to hydrate a `Stream`
```
Comment (thinkbeforecoding) [inline DOCUMENTATION.md:703]: Same

**Comment by thinkbeforecoding** [comment]
Seems to small comments changes, but good!
The Decider term start to be used in the eventsourcing community (at least on the DDD + EventModeling slacks)

File: DOCUMENTATION.md (hunk 2 above)
Comment (bartelink) [reply DOCUMENTATION.md:703]: that `Target` was actually way out of date, thanks for flagging 😊

File: DOCUMENTATION.md (hunk 1 above)
Comment (bartelink) [reply DOCUMENTATION.md:697]: Thanks for reviewing and catching; That was actually only the head of a seam, lots of fixes pushed 😊

**Pull request merged**